src/environment.cpp

```diff
@@ -191,6 +196,11 @@ void Environment::continueRaycast(RaycastState *state, PointedThing *result_p)
 			if (!is_valid_position)
 				continue;
 
+			// if take sense to use it
+			if ( (cy != y || cx != x || cz != z)
```

Comment:
How about just doing `state->m_iterator.m_current_node_pos != np`? Then we can also get rid of the three variables above.

Reply:
Or `v3s16 cp = state->m_iterator.m_current_node_pos;` if the goal was to move the to the stack for performance reasons.